specification/dataprotection/resource-manager/Microsoft.DataProtection/stable/2023-01-01/dataprotection.json

```diff
@@ -0,0 +1,7082 @@
+      "type": "object"
+    },
+    "ResourceGuard": {
+      "type": "object",
+      "properties": {
+        "provisioningState": {
+          "description": "Provisioning state of the BackupVault resource",
+          "enum": [
+            "Failed",
+            "Provisioning",
+            "Succeeded",
+            "Unknown",
+            "Updating"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "allowAutoApprovals": {
+          "description": "This flag indicates whether auto approval is allowed or not.",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "resourceGuardOperations": {
+          "description": "{readonly} List of operation details those are protected by the ResourceGuard resource",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ResourceGuardOperation"
+          },
+          "x-ms-identifiers": [],
+          "readOnly": true
+        },
+        "vaultCriticalOperationExclusionList": {
+          "description": "List of critical operations which are not protected by this resourceGuard",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "description": {
+          "description": "Description about the pre-req steps to perform all the critical operations.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ResourceGuardOperation": {
+      "description": "This class contains all the details about a critical operation.",
+      "type": "object",
+      "properties": {
+        "vaultCriticalOperation": {
+          "description": "Name of the critical operation.",
+          "type": "string",
+          "readOnly": true
+        },
+        "requestResourceType": {
+          "description": "Type of resource request.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ResourceGuardResource": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/DppBaseTrackedResource"
+        }
+      ],
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ResourceGuard",
+          "description": "ResourceGuardResource properties"
+        }
+      }
+    },
+    "ResourceGuardResourceList": {
+      "description": "List of ResourceGuard resources",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/DppTrackedResourceList"
+        }
+      ],
+      "properties": {
+        "value": {
+          "description": "List of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ResourceGuardResource"
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "SubscriptionId": {
```

Comment:
You won't need these definitions if you refer to common-types everywhere instead

Reply:
I've addressed and removed.